Replace SHA-512 hashing function with the ~10 times faster BLAKE3.

GoodByeZeroNet opened the pull request:
![blake3](https://user-images.githubusercontent.com/63570881/79089321-44d88280-7d35-11ea-86ae-041822df95e9.png)


BLAKE3 is a cryptographic hash function that is:

- **Much faster** than MD5, SHA-1, SHA-2, SHA-3, and BLAKE2.
- **Secure**, unlike MD5 and SHA-1. And secure against length extension,
  unlike SHA-2.
- **Highly parallelizable** across any number of threads and SIMD lanes,
  because it's a Merkle tree on the inside.
- Capable of **verified streaming** and **incremental updates**, again
  because it's a Merkle tree.
- A **PRF**, **MAC**, **KDF**, and **XOF**, as well as a regular hash.
- **One algorithm with no variants**, which is fast on x86-64 and also
  on smaller architectures.

The [chart below](https://github.com/BLAKE3-team/BLAKE3-specs/blob/master/benchmarks/bar_chart.py) shows BLAKE3's performance on modern server hardware, an Intel Cascade
Lake-SP 8275CL processor:
​
![speed](https://user-images.githubusercontent.com/63570881/79089128-77ce4680-7d34-11ea-89b9-cdd0885ff8ef.png)

BLAKE3 paper: https://github.com/BLAKE3-team/BLAKE3-specs/blob/master/blake3.pdf

---

**This is my last attempt to contribute to this repository** and the third time I changed the requirements.txt to include only gevent>=1.4.0! If this pull request is not going to be merged than this is going to be fastest fork of "ZeroNet". For that matter I going to rename it and will be not called "ZeroNet" anymore! 

I would like to give a huge thanks to @imachug and @filips123 for they great and hard work on this software, keep up the good work guys! :hugs: 

I wish that not @HelloZeroNet (@shortcutme) owns this repository. :cry:  It would be in better hands with @imachug for example. @imachug is way more respectful and active than the current "owner" :rage:.

Commit:
Update requirements.txt

Commit:
Update BigfilePlugin.py

Commit:
Update ContentManager.py

Commit:
Update CryptHash.py

Commit:
Update SiteStorage.py

Commit:
Update UiWebsocket.py

Commit:
Update Config.py

Commit:
Update main.py

Commit:
Update .travis.yml

purplesyringa commented:
Performance is a problem for ZeroNet given that it's written in Python. Hashes are checked really often (see a recent "Eternal Word" issue), so making them faster is important. BLAKE3 looks like a good alternative, but changing the hash function without backward compatibility is wrong. We could add both `blake3` and `sha512` fields when signing and check `blake3` on new versions and `sha512` on old versions on verifying.

ghost commented:
@imachug I'm not in favor of backward compatibility, at least not creating or signing sites with sha512.  Downloading the content which hashed by sha512 is enough.
Creating, signing new content should be hashed using blake3.
```
                else:  # Backward compatibility
                    hash_type = "sha1"
```
until now this :point_up: was in ContentManager.py and **turned out that is useless**, so I removed. Adding back sha512 instead of sha1 will still fail. There was no backward compatibility at all using this code.

At this point I see that this is going to be the fastest fork of ZeroNet. The @HelloZeroNet repository comes not even close to the speed of this fork with blake3.

Commit:
Update ContentManager.py

No need for "- ZeroNet_" in title

Commit:
Merge pull request #1 from peer-to-peer-network/peer-to-peer-network-patch-1

Update ContentManager.py

Commit:
Update ContentManager.py

Commit:
Update ContentManager.py

Everyone should resign they sites, no need for backward compatibility. :roll_eyes:
